Remove needless global vars and sys.argv usage

dimaka-wix opened the pull request:
Take a look just at the last commit (#b4ce9f2)

dimaka-wix committed:
Change max msg length

dimaka-wix committed:
Rename method extract... to get...

dimaka-wix committed:
Refactor the code according to the pull request notes of the team

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
-VIOLET = DEFAULT+'\033[35m'
-YELLOW = DEFAULT+'\033[33m'
+
+__all__ = ["main"]
```
Please add space below the `__all__`

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
         print(f"\n{RED}\
-ERROR: file '{path}' not found!\n{YELLOW}\
-HINT:  the commit message is usually saved in {COMMIT_EDITMSG}\n{DEFAULT}")
+ERROR: the path '{path}' not found!\n{YELLOW}\
+HINT:  the commit message is usually saved in {COMMIT_EDITMSG}{OFF}\n")
```
In this scenario, you can use string.format() instead of using the f string because in this way it's easier to see all the variables and also it can be less written in one line, so you won't need to do all the formating yourself with all the '/' and the new lines.

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
+        print(f"{RED}ERROR: capitalise {section}!{OFF}\n")
+        is_valid = False
     if not is_valid_prefix:
         nl = "...\n       "
```
You can use \t (tab)

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
-ERROR: wrong {segment} prefix!\n{YELLOW}\
-HINT:  you can add new prefixes as an {BLUE}args: {YELLOW}in {BLUE}\
-.pre-commit-config.yaml\n{YELLOW}\
+ERROR: wrong {section} prefix!\n{YELLOW}\
+HINT:  you can add new prefixes as an {BLUE}args: {YELLOW}in {BLUE}.pre-commit-config.yaml\n{YELLOW}\
```
Same as above

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
+
+    The example follows chaos-hub team commit rules.
+    """
+    print(f"{GREEN}\n\
+EXAMPLE:\n\
+  Refactor foo function in ...\n{BLUE}\
+<body is optional, adding it leave an empty line here>\n{GREEN}\
+  - Fix ...\n\
+  - Add ...\n\
+  - Remove ... \n{YELLOW}\
+HINT: to read chaos-hum team rules visit: {BLUE}{GITHUB_LINK}{OFF}\n")
```
Same as above

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
     """
-    global IS_EXIT_1
+    is_valid = True
     words = msg.strip().split()
```
Check if there are msg, if no, you need to print it and exit

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
     """
-    global IS_EXIT_1
+    is_valid = True
     if len(msg.splitlines()) > 1:
```
Check if msg exist

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
+    Returns:
+        bool: The result of the validation.
     """
     subj = msg.splitlines()[0]
```
Check if msg exist

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
-    is_valid_prefix = msg.lstrip().startswith(tuple(PREFIXES))
+    global default_prefixes
+    is_valid = True
+    is_valid_prefix = msg.lstrip().startswith(tuple(default_prefixes))
```
Same here with msg

roeiK-wix commented on src/main/cli.py:
```diff
@@ -6,280 +6,224 @@
     """
-    global IS_EXIT_1
+    is_valid = True
     if msg.rstrip().endswith("."):
```
Same here

dimaka-wix replied:
If the msg is empty the logic of the hook is not starts

dimaka-wix committed:
have a .

dimaka-wix committed:
Refactor the code
- Change validate functions to return string of detected errors
- Remove function

alexander-gorelik commented on README.md:
````diff
@@ -15,9 +15,9 @@ Add this to your ```.pre-commit-config.yaml```
     -   id: commit-msg-hookcs
-        # args is optional, the default length limit is 72
+        # add additional valid prefixes(optional)
+        args: [--prefix, Prefix1, Prefix2, ...]
````
Remove the test  Prefix1

dimaka-wix committed:
Update README
